### PR TITLE
NEXUS-21949: Remove v1 from the components and assets url in Conan Plugin

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-21949

### CODE REVIEW

File: nexus-repository-conan-it/src/test/resources/conan/download_urls [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-"conaninfo.txt": "http://127.0.0.1:57777/v1/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conaninfo.txt",
-"conan_package.tgz": "http://127.0.0.1:57777/v1/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conan_package.tgz",
-"conanmanifest.txt": "http://127.0.0.1:57777/v1/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conanmanifest.txt"
+"conaninfo.txt": "http://127.0.0.1:57777/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conaninfo.txt",
+"conan_package.tgz": "http://127.0.0.1:57777/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conan_package.tgz",
+"conanmanifest.txt": "http://127.0.0.1:57777/conans/vthiery/jsonformoderncpp/3.7.0/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/conanmanifest.txt"
 }
```
Comment (chernikovmaksym) [inline nexus-repository-conan-it/src/test/resources/conan/download_urls]: new line

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java [hunk 2]
```diff
@@ -41,43 +38,30 @@
...
-
-  public final Handler uploadConanPackage = context -> upload(context, "conan_package.tgz");
-
-  public final Handler uploadConanSources = context -> upload(context, "conan_sources.tgz");
-
-  public final Handler uploadConanExport = context -> upload(context, "conan_export.tgz");
-
-  private Response upload(final Context context, final String filename) throws IOException {
-
+  public final Handler uploadContentHandler = context -> {
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: I think it's better to put documentation above the variable. Look s a bit weird

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java [hunk 3]
```diff
@@ -112,14 +97,24 @@ private Response upload(final Context context, final String filename) throws IOE
...
+    String path = getHostedAssetPath(coord);
 
     return context.getRepository()
             .facet(ConanHostedFacet.class)
             .getPackageSnapshot(path, context);
   };
 
-  private String getAssetPath(final ConanCoords coord) {
-    return V1_CONANS + getPath(coord);
+  // TODO will be removed
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: Add ticket number or any additional info not to forget about this `TODO`

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java [hunk 4]
```diff
@@ -112,14 +97,24 @@ private Response upload(final Context context, final String filename) throws IOE
...
+  public static String getHostedAssetPath(final ConanCoords coord) {
+    return getHostedAssetPath(coord, null);
+  }
+
+  // TODO assetKind will be required
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: Add ticket number or any additional info not to forget about this `TODO`

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/utils/ConanFacetUtils.java [hunk 5]
```diff
@@ -39,6 +39,14 @@
 {
   public static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1, SHA256);
 
+  public static final String PACKAGE_SNAPSHOT_IDENTIFIER = "packages";
+
+  public static boolean isPackageSnapshot(String path) {
+    String[] args = path.split("/");
+    String expectPackage = args[args.length - 2];
```
Comment (chernikovmaksym) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/utils/ConanFacetUtils.java]: Maybe use a pattern here?

File: nexus-repository-conan-it/src/test/resources/conan/download_urls (hunk 1 above)
Comment (mlukaretkyi) [reply nexus-repository-conan-it/src/test/resources/conan/download_urls]: fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/91/commits/c39317f5cef05bd0906093aed821cbcb993407a9

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java (hunk 2 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/91/commits/c39317f5cef05bd0906093aed821cbcb993407a9

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java (hunk 3 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/91/commits/c39317f5cef05bd0906093aed821cbcb993407a9

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java (hunk 4 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/v1/HostedHandlers.java]: fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/91/commits/c39317f5cef05bd0906093aed821cbcb993407a9

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/common/v1/ConanRoutes.groovy [hunk 6]
```diff
@@ -157,12 +195,36 @@ class ConanRoutes
...
+                new TokenMatcher("{other:.*}/${STANDARD_FORM}/packages/{sha:.+}/conanmanifest.txt"),
+                new TokenMatcher("{other:.*}/${STANDARD_FORM}/conanmanifest.txt")
+            )
+        )
+    )
+  }
+
   private static TokenMatcher conanManifestMatcher() {
-    new TokenMatcher("/{path:.*}/${STANDARD_FORM}/conanmanifest.txt")
+    new TokenMatcher("{other:.*}/${STANDARD_FORM}/conanmanifest.txt")
```
Comment (igorudovika) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/common/v1/ConanRoutes.groovy]: Other looks like unclear. It is better to name it something like: api_prefix

Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/common/v1/ConanRoutes.groovy]: not a bad idea.
Fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/91/commits/d633930a9a11be20a74dc8463401b809b5bfe5bd

**Pull request merged**